pkg/cloud/aliyunprovider.go

```diff
@@ -0,0 +1,828 @@
+			return err
+		}
+		aliyun.Pricing[lookupKey] = pricingObj
+	}
+
+	// TO-DO: PV pricing
+	// //get pvList ultimately from aliyun cloud provider and resemble data from the pvtype to
+	// // Hardcodedk8sNodeDiskStruct
+	// pvList := aliyun.Clientset.GetAllPersistentVolumes()
+
+	// pvList := []*Hardcodedk8sNodeDiskStruct{}
+	// pvList = append(pvList, &Hardcodedk8sNodeDiskStruct{
+	// 	DiskType:         "data",
+	// 	DiskCategory:     "cloud",
+	// 	PerformanceLevel: "",
+	// 	RegionID:         "cn-hangzhou",
+	// 	PriceUnit:        "Hour",
+	// 	SizeInGiB:        60,
+	// 	ProviderID:       "Ali-XXX-pv-01",
+	// }, &Hardcodedk8sNodeDiskStruct{
+	// 	DiskType:         "data",
+	// 	DiskCategory:     "cloud",
+	// 	PerformanceLevel: "P1",
+	// 	RegionID:         "cn-hangzhou",
+	// 	PriceUnit:        "Hour",
+	// 	SizeInGiB:        40,
+	// 	ProviderID:       "Ali-XXX-pv-01",
+	// })
+
+	// for _, pv := range pvList {
+	// 	if client, ok = aliyun.clients[pv.RegionID]; !ok {
+	// 		client, err = sdk.NewClientWithAccessKey(pv.RegionID, aak.AccessKeyId, aak.AccessKeySecret)
+	// 		if err != nil {
+	// 			return fmt.Errorf("access key provided does not have access to location %s", pv.RegionID)
+	// 		}
+	// 		aliyun.clients[pv.RegionID] = client
+	// 	}
+	// 	signer = signers.NewAccessKeySigner(aak)
+	// 	pricingObj, err = processDescribePriceAndCreateAliyunPricing(client, pv, signer)
+	// 	lookupKey, err = determineKeyForPricing(pv)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+	// 	aliyun.Pricing[lookupKey] = pricingObj
+	// }
+	// log.Infof("Length of pricing is %d", len(aliyun.Pricing))
+	// log.Infof("random value is %v", aliyun.Pricing[lookupKey])
+	return nil
+}
+
+// AllNodePricing returns all the billing data for nodes and pvs
+func (aliyun *Aliyun) AllNodePricing() (interface{}, error) {
+	aliyun.DownloadPricingDataLock.RLock()
+	defer aliyun.DownloadPricingDataLock.RUnlock()
+	return aliyun.Pricing, nil
+}
+
+// NodePricing gives a specific node for the key
+func (aliyun *Aliyun) NodePricing(key Key) (*Node, error) {
+	aliyun.DownloadPricingDataLock.RLock()
+	defer aliyun.DownloadPricingDataLock.RUnlock()
+
+	// Get node features for the key
+	keyFeature := key.Features()
+
+	pricing, ok := aliyun.Pricing[keyFeature]
+	if !ok {
+		log.Infof("Node pricing information not found for node with feature: %s", keyFeature)
+		return &Node{}, nil
+	}
+
+	log.Infof("returing the node price for the node with feature: %s", keyFeature)
+	return pricing.Node, nil
+}
+
+// PVPricing gives a specific PV price for the PVkey
+func (aliyun *Aliyun) PVPricing(pvk PVKey) (*PV, error) {
+	aliyun.DownloadPricingDataLock.RLock()
+	defer aliyun.DownloadPricingDataLock.RUnlock()
+
+	keyFeature := pvk.Features()
+
+	pricing, ok := aliyun.Pricing[keyFeature]
+
+	if !ok {
+		log.Infof("Persistent Volume pricing not found for PV with feature: %s", keyFeature)
+		return &PV{}, nil
+	}
+
+	log.Infof("returing the PV price for the node with feature: %s", keyFeature)
```

Comment:
Similar log leveling, also typo
```suggestion
	log.Infof("returning the PV price for the node with feature: %s", keyFeature)
```